front/app/containers/Admin/formBuilder/components/FormBuilderTopBar/index.tsx

```diff
@@ -28,20 +29,26 @@ import { FormattedMessage } from 'utils/cl-intl';
 // routing
 import clHistory from 'utils/cl-router/history';
 import { useParams } from 'react-router-dom';
+
+// services
 import {
   IFlatCustomField,
   updateFormCustomFields,
 } from 'services/formCustomFields';
 
 interface FormBuilderTopBarProps {
-  formCustomFields: IFlatCustomField[] | undefined | Error;
+  control: Control;
 }
 
-const FormBuilderTopBar = ({ formCustomFields }: FormBuilderTopBarProps) => {
+const FormBuilderTopBar = ({ control }: FormBuilderTopBarProps) => {
   const localize = useLocalize();
   const { projectId } = useParams() as { projectId: string };
   const project = useProject({ projectId });
   const [loading, setLoading] = useState(false);
+  const formCustomFields: IFlatCustomField[] = useWatch({
```

Comment:
You can use watch directly from the `useFormContext` hook - no need to pass control as a prop and use the `useWatch` hook.